### ISSUE TITLE
Unable to start anbox on Ubuntu Mate 16.10

### ISSUE THREAD

**Issue by greenmanspirit** [anbox/anbox#4]
Make sure you are running the latest version of Anbox before reporting an issue.

**Anbox release (`anbox version`) and system info (`anbox system-info`):**
version: 0.1.0
os:
  name: Ubuntu Core
  version: 16
  snap-based: true
kernel:
  version: Linux version 4.8.0-46-generic (buildd@lcy01-16) (gcc version 6.2.0 20161005 (Ubuntu 6.2.0-5ubuntu12) ) #49-Ubuntu SMP Fri Mar 31 13:57:14 UTC 2017
  binder: true
  ashmem: true
graphics:
  egl:
    vendor: n/a
    version: n/a
    extensions: []
  gles2:
    vendor: n/a
    vendor: n/a
    extensions: []

**Description of the problem:**
Anbox session-manager does not start with "systemctl --user start anbox".

**Expected:**
Anbox session-manager to start

**Additional info:**
Systemctl --user status anbox
● anbox.service - Anbox session manager
   Loaded: loaded (/home/adam/.config/systemd/user/anbox.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2017-04-11 14:38:29 EDT; 1min 41s ago
  Process: 5666 ExecStart=/snap/bin/anbox session-manager (code=exited, status=1/FAILURE)
 Main PID: 5666 (code=exited, status=1/FAILURE)

Apr 11 14:38:29 imperio systemd[2629]: Started Anbox session manager.
Apr 11 14:38:29 imperio /snap/bin/anbox[5666]: cmd.go:111: DEBUG: restarting into "/snap/core/current/usr/bin/snap"
Apr 11 14:38:29 imperio systemd[2629]: anbox.service: Main process exited, code=exited, status=1/FAILURE
Apr 11 14:38:29 imperio systemd[2629]: anbox.service: Unit entered failed state.
Apr 11 14:38:29 imperio systemd[2629]: anbox.service: Failed with result 'exit-code'.

Journalctl
Apr 11 14:38:29 imperio systemd[2629]: Started Anbox session manager.
Apr 11 14:38:29 imperio /snap/bin/anbox[5666]: cmd.go:111: DEBUG: restarting into "/snap/core/current/usr/bin/snap"
Apr 11 14:38:29 imperio audit[5666]: AVC apparmor="ALLOWED" operation="open" profile="snap.anbox.anbox" name="/proc/5666/cgroup" pid=5666 comm="anbox" requested_mask="r" denied_mask="r" fsuid=2211 ouid=2211
Apr 11 14:38:29 imperio audit[5666]: AVC apparmor="ALLOWED" operation="open" profile="snap.anbox.anbox" name="/proc/5666/mountinfo" pid=5666 comm="anbox" requested_mask="r" denied_mask="r" fsuid=2211 ouid=2211
Apr 11 14:38:29 imperio audit[5666]: AVC apparmor="ALLOWED" operation="open" profile="snap.anbox.anbox" name="/proc/5666/cgroup" pid=5666 comm="anbox" requested_mask="r" denied_mask="r" fsuid=2211 ouid=2211
Apr 11 14:38:29 imperio audit[5666]: AVC apparmor="ALLOWED" operation="connect" profile="snap.anbox.anbox" name="/run/anbox-container.socket" pid=5666 comm="anbox" requested_mask="wr" denied_mask="wr" fsuid=2211 ouid=0
Apr 11 14:38:29 imperio anbox[5666]: [EE 2017-04-11 18:38:29] [daemon.cpp:58@Run] Failed to initialize SDL: Failed to connect to the Mir Server
Apr 11 14:38:29 imperio kernel: audit_printk_skb: 144 callbacks suppressed
Apr 11 14:38:29 imperio kernel: audit: type=1400 audit(1491935909.714:2231): apparmor="ALLOWED" operation="open" profile="snap.anbox.anbox" name="/proc/5666/cgroup" pid=5666 comm="anbox" requested_mask="r" denied_mask="r" fsuid=2211 ouid=2211
Apr 11 14:38:29 imperio kernel: audit: type=1400 audit(1491935909.714:2232): apparmor="ALLOWED" operation="open" profile="snap.anbox.anbox" name="/proc/5666/mountinfo" pid=5666 comm="anbox" requested_mask="r" denied_mask="r" fsuid=2211 ouid=2211
Apr 11 14:38:29 imperio kernel: audit: type=1400 audit(1491935909.714:2233): apparmor="ALLOWED" operation="open" profile="snap.anbox.anbox" name="/proc/5666/cgroup" pid=5666 comm="anbox" requested_mask="r" denied_mask="r" fsuid=2211 ouid=2211
Apr 11 14:38:29 imperio kernel: audit: type=1400 audit(1491935909.714:2234): apparmor="ALLOWED" operation="connect" profile="snap.anbox.anbox" name="/run/anbox-container.socket" pid=5666 comm="anbox" requested_mask="wr" denied_mask="wr" fsuid=2211 ouid=0
Apr 11 14:38:29 imperio systemd[2629]: anbox.service: Main process exited, code=exited, status=1/FAILURE
Apr 11 14:38:29 imperio systemd[2629]: anbox.service: Unit entered failed state.
Apr 11 14:38:29 imperio systemd[2629]: anbox.service: Failed with result 'exit-code'.

**Comment by morphis** [comment]
The actual problem here is 

```
Apr 11 14:38:29 imperio anbox[5666]: [EE 2017-04-11 18:38:29] [daemon.cpp:58@Run] Failed to initialize SDL: Failed to connect to the Mir Server
```

I am not yet sure where this comes from. We don't do anything Mir specific in Anbox and are just using X11. It looks like SDL fails to talk to X11 and then tries the next one in the list (mir). See https://github.com/anbox/anbox/blob/master/src/anbox/ubuntu/platform_policy.cpp#L44

**Comment by greenmanspirit** [comment]
→ ANBOX_LOG_LEVEL=debug anbox session-manager
[II 2017-04-11 19:23:56] [session_manager.cpp:126@operator()] Detected properietary nvidia driver; forcing use of the host EGL driver.
[EE 2017-04-11 19:23:57] [Renderer.cpp:100@initialize] Failed to Initialize backend EGL display
^C[II 2017-04-11 19:26:44] [session_manager.cpp:114@operator()] Signal 2 received. Good night.

**Comment by keithzg** [comment]
Funny enough, on my Kubuntu 16.04 install I've tried to use the systemd unit file provided by the installation script and gotten the exact same complaint:

```
anbox[32070]: [EE 2017-04-12 02:40:19] [daemon.cpp:58@Run] Failed to initialize SDL: Failed to connect to the Mir Server
```

Yet, if I too run `ANBOX_LOG_LEVEL=debug anbox session-manager` (or even just `anbox session-manager`), instead things proceed!

```
keithzg@siddhartha:~/bin$ ANBOX_LOG_LEVEL=debug anbox session-manager
/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
[DD 2017-04-12 02:43:39] [gl_renderer_server.cpp:82@GLRendererServer] Using GLES-to-GL translator for rendering
[DD 2017-04-12 02:43:39] [Renderer.cpp:113@initialize] EGL doesn't support surfaceless context
[DD 2017-04-12 02:43:39] [Renderer.cpp:249@initialize] Successfully initialized EGL
[DD 2017-04-12 02:43:46] [session_manager.cpp:208@operator()] Android successfully booted
```

Of course my situation is otherwise quite unlike @greenmanspirit; I am running KDE on 16.04, and I'm using the non-proprietary AMD drivers. Perhaps that explains my success when manually invoking it, but the failure when running it as a systemd service is due to something shared?

**Comment by morphis** [comment]
> Funny enough, on my Kubuntu 16.04 install I've tried to use the systemd unit file provided by the installation script and gotten the exact same complaint:

I really wonder what is wrong here on Kubuntu. Need to try this one day but somehow certain things are missing when the session manager is started from the systemd unit and it can't contact X11.

Can you check the following: `$ export XAUTHORITY`
If that points to something in /tmp I know what the problem is.

**Comment by AryToNeX** [comment]
Using elementary OS 0.4 loki

Installed Anbox from installer.sh, rebooted, clicked the icon, it doesn't work.


```$ env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/anbox_anbox.desktop /snap/bin/anbox launch --package=org.anbox.appmgr --component=org.anbox.appmgr.AppViewActivity

[WW 2017-04-12 19:54:35] [launch.cpp:115@operator()] Stop waiting as we're already waiting for too long. Something is wrong
[WW 2017-04-12 19:54:35] [launch.cpp:116@operator()] with your setup and the container may have failed to boot.


$ systemctl status snap.anbox.container-manager.service

● snap.anbox.container-manager.service - Service for snap application anbox.container-manager
   Loaded: loaded (/etc/systemd/system/snap.anbox.container-manager.service; enabled; vendor preset: enabled)
   Active: active (running) since mer 2017-04-12 21:38:50 CEST; 16min ago
 Main PID: 1394 (anbox)
   CGroup: /system.slice/snap.anbox.container-manager.service
           ├─1394 /snap/anbox/15/usr/bin/anbox container-manager --data-path=/var/snap/anbox/common/ --android-image=/snap/anbox/15/android.img
           └─2759 dnsmasq -u dnsmasq --strict-order --bind-interfaces --pid-file=/run/anbox/dnsmasq.pid --dhcp-no-override --except-interface=lo --interface=anboxbr0 --dhcp-leasefile=/var/lib/anbox/dnsmasq.anboxbr0.leases --dhcp-authoritat

apr 12 21:39:06 anthony-PC snap[1394]: + exec /snap/anbox/15/usr/sbin/aa-exec -p unconfined -- /snap/anbox/15/bin/anbox-wrapper.sh container-manager --data-path=/var/snap/anbox/common/ --android-image=/snap/anbox/15/android.img
apr 12 21:39:11 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:39:18 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:39:34 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:39:39 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:39:44 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:39:54 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:39:59 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:40:18 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::
apr 12 21:48:34 anthony-PC dnsmasq-dhcp[2759]: RTR-ADVERT(anboxbr0) fd9d:e4dc:4e00:9e98::


$ initctl status anbox

initctl: Name "com.ubuntu.Upstart" does not exist


$ systemctl --user status anbox

● anbox.service - Anbox session manager
   Loaded: loaded (/home/anthony/.config/systemd/user/anbox.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since mer 2017-04-12 21:39:43 CEST; 16min ago
  Process: 3884 ExecStart=/snap/bin/anbox session-manager (code=exited, status=1/FAILURE)
 Main PID: 3884 (code=exited, status=1/FAILURE)

apr 12 21:39:43 anthony-PC systemd[3878]: Started Anbox session manager.
apr 12 21:39:43 anthony-PC /snap/bin/anbox[3884]: cmd.go:111: DEBUG: restarting into "/snap/core/current/usr/bin/snap"
apr 12 21:39:43 anthony-PC anbox[3884]: [EE 2017-04-12 19:39:43] [daemon.cpp:58@Run] Failed to initialize SDL: Failed to connect to the Mir Server
apr 12 21:39:43 anthony-PC systemd[3878]: anbox.service: Main process exited, code=exited, status=1/FAILURE
apr 12 21:39:43 anthony-PC systemd[3878]: anbox.service: Unit entered failed state.
apr 12 21:39:43 anthony-PC systemd[3878]: anbox.service: Failed with result 'exit-code'.
```

**Comment by acetheultimate** [comment]
Had same problem.
Workaroud:
`
you@pc:~$ anbox session-manager
`
Launch it from launcher!

**Comment by Pillar1989** [comment]
I had the same problem. It seem nobody run anbox session-manager successfull.

**Comment by edopudiyoraharjo** [comment]
i have same problem in Ubuntu 16.10 MATE but my session-manager is actived 
anbox session-manager
[EE 2017-04-15 04:26:31] [daemon.cpp:58@Run] bind: Address already in use

but if i launch anbox with anbox launch ->
anbox launch
[WW 2017-04-15 04:33:02] [launch.cpp:115@operator()] Stop waiting as we're already waiting for too long. Something is wrong
[WW 2017-04-15 04:33:02] [launch.cpp:116@operator()] with your setup and the container may have failed to boot.

**Comment by hahampis** [comment]
Hi Morphis, 

I am on Kubuntu 16.04 and have the same issue: 
Failed to initialize SDL: Failed to connect to the Mir Server

You mentioned XAUTHORITY:
~$ echo $XAUTHORITY
/tmp/xauth-1000-_0

It does point to someting in /tmp! Do you then know what the problem is? :)

**Comment by keithzg** [comment]
@morphis wrote:
> I really wonder what is wrong here on Kubuntu. Need to try this one day but somehow certain things are missing when the session manager is started from the systemd unit and it can't contact X11.
> 
> Can you check the following: $ export XAUTHORITY
> If that points to something in /tmp I know what the problem is

@hahampis wrote:
> You mentioned XAUTHORITY:
> ~$ echo $XAUTHORITY
> /tmp/xauth-1000-_0
> It does point to someting in /tmp! Do you then know what the problem is? :)

I'm afraid I'm going to have to complicate things here . . .

```
keithzg@siddhartha:~$ systemctl --user start anbox
keithzg@siddhartha:~$ systemctl --user status anbox
● anbox.service - Anbox session manager
   Loaded: loaded (/home/keithzg/.config/systemd/user/anbox.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2017-04-17 14:19:29 MDT; 2s ago
  Process: 6267 ExecStart=/snap/bin/anbox session-manager (code=exited, status=1/FAILURE)
 Main PID: 6267 (code=exited, status=1/FAILURE)

Apr 17 14:19:28 siddhartha systemd[6272]: Started Anbox session manager.
Apr 17 14:19:28 siddhartha /snap/bin/anbox[6267]: cmd.go:111: DEBUG: restarting into "/snap/core/current/usr/bin/snap"
Apr 17 14:19:29 siddhartha anbox[6267]: [EE 2017-04-17 20:19:29] [daemon.cpp:58@Run] Failed to initialize SDL: Failed to connect to the Mir Server
Apr 17 14:19:29 siddhartha systemd[6272]: anbox.service: Main process exited, code=exited, status=1/FAILURE
Apr 17 14:19:29 siddhartha systemd[6272]: anbox.service: Unit entered failed state.
Apr 17 14:19:29 siddhartha systemd[6272]: anbox.service: Failed with result 'exit-code'.
keithzg@siddhartha:~$ echo $XAUTHORITY
/home/keithzg/.Xauthority
```

I've even tried explicitly adding `Environment="XAUTHORITY=/home/keithzg/.Xauthority"` into the service unit file, to no avail.

**Comment by morphis** [comment]
The original issue described here is because we don't support the Nvidia driver yet. Duplicate of https://github.com/anbox/anbox/issues/11

**Comment by keithzg** [comment]
Umm, are you sure, @morphis? I definitely do not have any NVIDIA card even plugged into the machine in question, much less am I running those drivers. Should I be opening a new issue for my experience of this, then?

**Comment by hub2git** [comment]
Hi @keithzg, @morphis was addressing the original OP's issue
(@greenmanspirit). As @morphis wrote in
https://github.com/anbox/anbox/issues/4#issuecomment-296072566, "The
original issue described here..." This is why Simon closed this issue.

Maybe your issue has not been reported. May I suggest searching through the
issues first, including the closed ones. Then post a new issue if you don't
see a similar issue posted.

**Comment by keithzg** [comment]
@hub2git: I mean, "similar issue" that I saw _was_ this one, which is why I chimed in in the first place. But alrighty, I'll just create a new issue.

**Comment by hub2git** [comment]
@keithzg, I'd first check that your issue has not already been posted.
Likely, it has.

**Comment by keithzg** [comment]
@hub2git If it has, I don't see it. I tried searching for "systemd" in the open issues list and none of the ones open appear to be the same as mine. Seeing as I do _not_ experience issues when running anbox manually but _do_ when run as a systemd user unit, the usage of systemd would seem to be the key here.

It is possible, of course, that something else is the root cause. By the same token, however, that could be true of this case if the assumption that NVIDIA was at fault is incorrect. There is not much else for me to compare against in terms of issues; for instance this is one of only two cases mentioning the error message about Mir that I received, and the other is someone trying to run anbox under Wayland on Arch (neither of which applies to me) _and_ that one was closed as a dupe of #11 as well, which again doesn't apply to me since I am not running any NVIDIA hardware or drivers on this machine.

So you may think it was likely, but you'd have lost that bet.

**Comment by hub2git** [comment]
@keithzg. Feel free to post a unique issue. You mentioned searching through
the open-issues list. It seems you also looked through the closed issues.
Hope your issue and mine get fixed, as it's _ likely_ :laughing: you are as
excited as I am to use Anbox. (That's a bet I may make! :smile:)
